gravitee-apim-console-webui/src/components/notifications/notifications-list/notification-details/notifications-details-reusable.component.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, EventEmitter, Inject, Input, OnInit, Output } from "@angular/core";
+import { FormControl, FormGroup } from "@angular/forms";
+import { takeUntil, tap } from "rxjs/operators";
+import { Observable, Subject } from "rxjs";
+import { groupBy, map } from "lodash";
+import { StateService } from "@uirouter/angular";
+
+import { UIRouterState } from "../../../../ajs-upgraded-providers";
+import { SnackBarService } from "../../../../services-ngx/snack-bar.service";
+import { Hooks } from "../../../../entities/notification/hooks";
+import { Notifier } from "../../../../entities/notification/notifier";
+import { NotificationSettings } from "../../../../entities/notification/notificationSettings";
+
+export interface NotificationDetailsServices {
+  reference: {
+    referenceType: 'API' | 'APPLICATION' | 'PORTAL',
+    referenceId: 'DEFAULT' | 'APPLICATION'
+  };
+  update: (updatedNotification: NotificationSettings) => Observable<NotificationSettings>;
+}
+
+type CategoriesHooksVM = {
+  name: string;
+  hooks: (Hooks & { checked: boolean })[];
+}[];
+
+@Component({
+  selector: "notifications-details-reusable",
+  template: require("./notifications-details-reusable.component.html"),
+  styles: [require("./notifications-details-reusable.component.scss")]
+})
+export class NotificationsDetailsReusableComponent implements OnInit {
+  public notificationForm: FormGroup;
+  public routingPath: string;
+  public formInitialValues: unknown;
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  @Input() categoriesHooksVM: CategoriesHooksVM;
+  @Input() notifier: Notifier;
+  @Input() notificationSettings: NotificationSettings;
+  @Input() hooks: any;
```

Comment:
What do you think about taking care not to keep `any`  type in the code? 🙏

Reply:
done